### DISCUSSION TITLE
Error w/ Management API - m2m backend server

### DISCUSSION THREAD

**Discussion by LBF38** [logto-io/logto#5407] [Q&A] [answered]
Hi, I have been trying to [Interact with Management API](https://docs.logto.io/docs/recipes/interact-with-management-api/) for fetching user data and creating [User Profiles](https://docs.logto.io/docs/recipes/user-profile/). 
Unfortunately, I haven't succeeded and I'm stuck with the following error :

```json
{
  "message": "users",
  "users": {
    "message": "Unauthorized. Please check credentials and its scope.",
    "code": "auth.unauthorized",
    "data": {
      "code": "ERR_JWKS_NO_MATCHING_KEY",
      "name": "JWKSNoMatchingKey",
      "message": "no applicable key found in the JSON Web Key Set"
    }
  }
}
``` 

My setup is as follow:
- LogTo running in docker image and accessible at `http://localhost:3001` and `http://localhost:3002`
- My backend server registered as an API resource in LogTo and as an M2M app for `https://default.logto.app/api`, to get users' data.
   - the server also runs in a Docker container, everything is setup as a Docker compose file.
   - I use an Express app for the server, all in TypeScript.

Has anyone got an idea on what I'm doing wrong or what the solution is ?

Thanks in advance for your answers.

## Ressources 
Here are some links that I've found and that are related/similar to this issue but didn't answer my question:
- https://github.com/logto-io/logto/discussions/1971
- https://github.com/logto-io/logto/issues/5186

**Comment by simeng-li** [comment]
Did you assign the admin role to the m2m application?

**Reply by LBF38** [reply]
Yes, it has the admin role for the Management API : 
![image](https://github.com/logto-io/logto/assets/96347823/09670608-8c45-41c2-bff8-39c1897dd30a)

I think it might be an issue with Docker networks ? Or maybe the `iss` in the jwt token ? I don't know..

Here is the jwt token produced by the api call to get the M2M token : https://jwt.io/#debugger-io?token=eyJhbGciOiJFUzM4NCIsInR5cCI6ImF0K2p3dCIsImtpZCI6InZQeFlJUjR0QW52SkZIMWJOOU1xblRQX2R6TERfUEdJaHlCR01icURGcEkifQ.eyJqdGkiOiJBOEN1U3RFUnRESjlxbFVEbHdBM1AiLCJzdWIiOiJyOXcycDIweTJ0OTJwZ3ZkZjAwMHEiLCJpYXQiOjE3MDgwOTMxNDQsImV4cCI6MTcwODA5Njc0NCwic2NvcGUiOiJhbGwiLCJjbGllbnRfaWQiOiJyOXcycDIweTJ0OTJwZ3ZkZjAwMHEiLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjMwMDEvb2lkYyIsImF1ZCI6Imh0dHBzOi8vZGVmYXVsdC5sb2d0by5hcHAvYXBpIn0.mj2_ln7O-bnl1HW8RuAVgRgbY22i1vWMflNak5xfsZpiio1rUx4fDao1iuYmeI3MDS67AbGhIijRfJ1DlO_u5vNu5wq0QOBX2D_OK7L4cDOE1sodk4Rno5bNzy4ydz5q

The code used to produce this code is exactly the same as the one from [Logto's documentation](https://docs.logto.io/sdk/m2m/).
I use `https://default.logto.app/api` as the resource for accessing every api endpoints.

**Reply by LBF38** [reply]
Here is a link to a reproduction of the issue I encountered : https://github.com/LBF38/logto_repro
Everything is in the repo. You will have to create a m2m app in the logto instance before running the bash script.
Thx

**Reply by LBF38** [reply]
Hi @simeng-li , do you have any idea to solve this issue ? 
Thank you in advance for your help

**Reply by simeng-li** [reply, marked as answer]
😆  replace the `https://default.logto.app/api` with your actual local service endpoint. e.g. `http://localhost:3001/api/users`.

**Reply by LBF38** [reply]
that works better ! 😄
